lib/msf/core/post/windows/wmic.rb

```diff
@@ -22,6 +22,40 @@ def initialize(info = {})
                      ], self.class)
   end
 
+  def gwmi_query(query, server=datastore['RHOST'], filter='', namespace='')
+    extapi = load_extapi
+
+    result_text = ""
+
+    psh_cmd = "Get-WmiObject -Query '#{query}' "
+    if (namespace.nil? or namespace.empty?) == false
+      psh_cmd.concat("-Namespace #{namespace}")
+    end
+    if (filter.nil? or filter.empty?) == false
+      psh_cmd.concat("| select #{filter}")
+    end
+
+    vprint_status("[#{server}] #{psh_cmd}")
+
+    encoded_psh_cmd = Rex::Powershell::Command.encode_script(psh_cmd)
+    qwmicmd = "powershell.exe -encodedcommand #{encoded_psh_cmd} -nop -w hidden"
```

Comment:
It may or may not be a good idea to execute `raise "Powershell is required" if !have_powershell?` first. Although executing this for every query seems a little excessive...

`have_powershell?` requires the `Msf::Post::Windows::Powershell` mixin, and is defined in [lib/msf/core/post/windows/powershell.rb](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/post/windows/powershell.rb).

Reply:
Yeah it seems a bit overkill to implement for this module. One option would be to add a `check` method that checks if `powershell` is available and in `%PATH%` (`have_powershell?`).

Alternatively, the `session` object could be updated with support for a `have_powershell` or `powershell?` property, but that's outside the scope of this PR.

Marking as resolved.